ko/sql/dblink.rst

```diff
@@ -272,122 +220,370 @@ cubrid_gateway.conf 파일
+	
+Oracle 환경설정
+----------------------------
+
+DBLink에서 Oracle을 사용하기 위해서는 Oracle Instant Client 설치 및 설정, 연결 정보 설정, Oracle Database 환경변수 설정 및 게이트웨이 설정을 반드시 해야 한다.
+
+
+**Oracle Database 환경변수 설정**
+
+
+Oracle database server 에 아래의 환경변수를 설정해야 한다.
+
+
+::
+	
+	export ORACLE_SID=XE
+	export ORACLE_BASE=/u01/app/oracle
+	export ORACLE_HOME=$ORACLE_BASE/product/11.2.0/xe
+	export PATH=$ORACLE_HOME/bin:$PATH
+
+
+* ORACLE_SID는 시스템 식별자이다.
+* ORACLE_BASE은 오라클 기본 디렉토리 구조이다.
+* ORACLE_HOME은 오라클 데이터베이스가 설치된 경로이다.	
+
+**오라클 인스턴트 클라이언트 ODBC 설치**
+
+Oracle Instant Client 다운로드 사이트에서 ODBC Package와 Basic Package 다운받아 동일한 디렉토리에 압축을 풉니다.
+
+::
+    
+	unzip instantclient-basic-linux.x64-11.2.0.4.0.zip
+	unzip instantclient-odbc-linux.x64-11.2.0.4.0.zip
+
+Oracle Instant Client 다운로드 사이트: https://www.oracle.com/database/technologies/instant-client/downloads.html
+
+
+**오라클 인스턴트 클라이언트 환경변수 설정**
+
+export ORACLE_INSTANT_CLIENT=/home/user/oracle/instantclient  
+export PATH=$ORACLE_INSTANT_CLIENT:$PATH
+export LD_LIBRARY_PATH=$ORACLE_INSTANT_CLIENT:$LD_LIBRARY_PATH
+
+
+.. _tnsnames-info:
+
+**Oracle Database에 연결을 위한 연결정보 설정**
+
+Oracle Database에 연결을 하기 위해서는 연결정보를 가지고 있는 tnsnames.ora 파일을 수정해야 한다.
+아래의 기본 형식에 HOST, PORT, SERVICE_NAME 이 세 항목에 연결정보를 작성해야 한다.
+연결정보를 작성한 tnsnames.ora 파일은 TNS_ADMIN 환경변수에서 디렉토리 경로를 설정해야 한다. 
+TNS_ADMIN설정 방법은 "TNS_ADMIN 환경변수 설정" 참고한다.
+
+
+tnsnames.ora 파일의 기본 형식
+
+::
+	
+	net_service_name =
+	  (DESCRIPTION=
+		(ADDRESS = (PROTOCOL = TCP)(HOST = xxx.xxx.xxx.xxx)(PORT = 1521)
+	  )
+	  (CONNECT_DATA =
+		(SERVICE_NAME=service_name)
+	  )
+	)
+
+
+* net_service_name: 데이터베이스 연결을 위한 네트 서비스 이름이며, connection url의 db_name에 사용하는 이름이다.
+* HOST: 데이터베이스에 연결하려는 IP 주소 또는 서버 이름이다.
+* PORT: 연결에 필요한 포트이다. 대부분의 경우 기본 포트는 1521이다.
+* service_name: 연결하려는 데이터베이스의 이름이다.
+
+
+.. note::
+    
+        참고로, net_service_name 이 중복으로 작성이 되어도 에러가 발생되지 않는다. 하지만 중복된 다른 서버에 연결될 수 있으므로, net_service_name 이 반드시 중복되지 않게 설정해야 한다.
+
+
+.. _tns_admin-info:
+
+**TNS_ADMIN 환경변수 설정**
+
+TNS_ADMIN는 tnsnames.ora 파일이 있는 디렉토리 경로를 가리킨다.
+만약 /home/user/myconfigs 에 tnsnames.ora 파일이 있다면 아래와 같이 설정 할 수 있다.
+
+::
+	
+	export TNS_ADMIN=/home/user/myconfigs
+
 
-동일기종의 Brokers와 이기종의 GATEWAY 설정을 했다면, 데이터베이스의 정보를 조회하기 위한 Query문 작성 방법에 대해서 알아본다.
 
-데이터 조회를 위한 DBLINK Query문 작성 방법 두가지가 있다.
+**Oracle을 위한 cubrid_gataway.conf 설정**
+
+게이트웨이에서 oracle에 연결하기 위해서는 아래와 같이 몇 가지 설정이 필요하다.
+
+자세한 :ref:`게이트웨이 설정 파일 <gatewayconf-info>`\ 을 참고한다.
+
+게이트웨이는 oracle에 연결하기 위해서 tnsnames.ora 의 정보를 이용하기 때문에  CGW_LINK_SERVER_IP, CGW_LINK_SERVER_PORT 는 작성하지 않아도 된다. 해당 정보를 작성한 경우에도 게이트웨이는 참조하지 않는다.
 
-첫째, FROM절에 DBLINK 구문을 작성하여 타 데이터베이스의 정보를 조회하는 방법
-아래의 Query문은 IP 192.168.0.1의 타 데이터베이스의 remote_t 테이블 정보를 조회하는 Query문이다.
 
 ::
     
-	SELECT * FROM DBLINK ('192.168.0.1:53000:demodb:user:password:','SELECT col1, col2 FROM remote_t') AS t(col1 int, col2 varchar(32));
+	APPL_SERVER              	=CAS_CGW
+			.
+			.
+			.
+	CGW_LINK_SERVER		        =ORACLE
+	CGW_LINK_SERVER_IP      	=localhost
+	CGW_LINK_SERVER_PORT    	=1521

```

Comment:
Is it right? You said CGW_LINK_SERVER_IP, CGW_LINK_SERVER_PORT are not used in case of Oracle.

Reply:
It is correct that these two parameters are not used. But the parameters do exist. 
So, I explained how to use it in the manual. (line 456)

Reply:
I will delete it. thank you